server/server.go

```diff
@@ -120,6 +123,13 @@ func (s *server) ListenAndServe() error {
 	return nil
 }
 
+func (s *server) Serve(l net.Listener) error {
+	s.address = l.Addr().String()
+	s.listener = l
+	go s.serve()
+	return nil
```

Comment:
It's not necessary in this case, but I can imagine we might have other implementations of the `Server` interface that might return an error. This way we wouldn't have to change the function signature in the future. I don't feel too strongly either way though.